src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs

```diff
@@ -29,16 +30,16 @@ IActionResult GetInvalidModelStateResponse(ActionContext context)
             {
                 var errorDetails = _errorDescriptionFactory.CreateErrorDescription(
                     context.ActionDescriptor, 
-                    new ValidationProblemDetails(context.ModelState));
-
-                return new BadRequestObjectResult(errorDetails)
-                {
-                    ContentTypes =
-                    {
-                        "application/problem+json",
-                        "application/problem+xml",
-                    },
-                };
+                    context.ModelState);
+
+                var result = (errorDetails is ModelStateDictionary modelState) ?
+                    new BadRequestObjectResult(modelState) :
+                    new BadRequestObjectResult(errorDetails);
+
+                result.ContentTypes.Add("application/problem+json");
+                result.ContentTypes.Add("application/problem+json");
```

Comment:
Was this intentional?

Reply:
If this is still here, it's wrong. We're added a class definition for `problem` but we're not actually using it. Thanks for catching this.

Reply:
I spoke to @Eilon about this. I missed changing this (I thought I did) and he didn't think it met the ask-mode bar to fix at this point.